### PR TITLE
re did wocutoff, and lots and lots of refactoring

### PR DESCRIPTION
**Pull request by TordJoranger**

### CODE REVIEW

File: Core/Services/DbStatusFeederService.cs [hunk 1]
```diff
@@ -21,58 +19,47 @@ public class DbStatusFeederService : IDbStatusFeederService
...
             foreach (var metric in metrics)
             {
-                var props = new Dictionary<string, string>();
-                props.Add("UserName", metric.UseName);
-                props.Add("Program", metric.Program);
-                props.Add("SID", metric.Sid.ToString());
-                props.Add("Serial", metric.Serial.ToString());
+                var props = new Dictionary<string, string>
+                {
+                    { "UserName", metric.UseName },
```
Comment (jsoi-eq) [inline Core/Services/DbStatusFeederService.cs:47]: OCD: metric.UserName

File: FamFeederFunction/Functions/FamFeederFunction.cs [hunk 2]
```diff
@@ -83,23 +83,34 @@ public FamFeederFunction(IFamFeederService famFeederService)
...
-    public static async Task<List<string>> RunOrchestrator(
+    public static async  Task<List<string>> RunOrchestrator(
         [OrchestrationTrigger] IDurableOrchestrationContext context)
     {
         var param = context.GetInput<QueryParameters>();
-        var results = new List<string>();
+        var results = new List<Task<string>>();
         if (param.PcsTopic == PcsTopic.WorkOrderCutoff)
         {
             var months = new List<string> { "01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12" };
```
Comment (jsoi-eq) [inline FamFeederFunction/Functions/FamFeederFunction.cs:93]: For greater readability, move content of if block to separate method.

File: FamFeederFunction/Functions/FamFeederFunction.cs [hunk 3]
```diff
@@ -83,23 +83,34 @@ public FamFeederFunction(IFamFeederService famFeederService)
...
             foreach (var cutoffInput in months.Select(m => (param.Plant, m)))
             {
-                results.Add(await context.CallActivityAsync<string>("RunWoCutoffFeeder", cutoffInput));
+                var woForMonthTask = context.CallSubOrchestratorAsync<string>("RunWoCutoffFeeder", cutoffInput);
+                results.Add(woForMonthTask);
+
```
Comment (jsoi-eq) [inline FamFeederFunction/Functions/FamFeederFunction.cs:98]: Remove extra line break

File: Infrastructure/Repositories/DbStatusRepository.cs [hunk 4]
```diff
@@ -19,26 +20,43 @@ public DbStatusRepository(AppDbContext context)
...
+            while (await result.ReadAsync())
+            {
+                metrics.Add(new MetricDto()
+                {
+                    UseName = result.IsDBNull(0) ? "" : result.GetString(0),
+                    Program = result.IsDBNull(1) ? "" : result.GetString(1),
+                    Sid = result.GetInt32(2),
+                    Serial = result.GetInt32(3),
+                    Name = result.IsDBNull(4) ? "" : result.GetString(4),
+                    Value = result.GetInt32(5),
```
Comment (jsoi-eq) [inline Infrastructure/Repositories/DbStatusRepository.cs:47]: 5 is a magic number?

File: Infrastructure/Repositories/PlantRepository.cs [hunk 5]
```diff
@@ -12,18 +13,32 @@ public class PlantRepository : IPlantRepository
...
+        {
+            await using var command = dbConnection.CreateCommand();
+            command.CommandText = "Select projectschema from projectschema";
+            await using var result = await command.ExecuteReaderAsync();
+            var plants = new List<string>();
 
-        return plants;
+            while (await result.ReadAsync())
+            {
+                plants.Add((string)result[0]);
```
Comment (jsoi-eq) [inline Infrastructure/Repositories/PlantRepository.cs:31]: 0 is magic number?

File: Infrastructure/Repositories/WorkOrderCutoffRepository.cs [hunk 6]
```diff
@@ -2,26 +2,50 @@
...
+            await _context.Database.OpenConnectionAsync();
+        }
+        try
+        {
+            await using var result = await command.ExecuteReaderAsync();
+            var entities = new List<string>();
+
+            while (await result.ReadAsync())
+            {
+                entities.Add((string)result[0]);
```
Comment (jsoi-eq) [inline Infrastructure/Repositories/WorkOrderCutoffRepository.cs:36]: Magic number?

File: Infrastructure/Repositories/PlantRepository.cs (hunk 5 above)
Comment (TordJoranger) [reply Infrastructure/Repositories/PlantRepository.cs:31]: I'd argue that index 0 is not so magic when it comes to arrays :P

File: Infrastructure/Repositories/WorkOrderCutoffRepository.cs (hunk 6 above)
Comment (jsoi-eq) [reply Infrastructure/Repositories/WorkOrderCutoffRepository.cs:36]: Is array, closing comment...

File: Core/Services/DbStatusFeederService.cs [hunk 7]
```diff
@@ -21,58 +19,47 @@ public class DbStatusFeederService : IDbStatusFeederService
         _famFeederOptions = famFeederOptions.Value;
     }
 
-    public async Task<string> RunFeeder(ILogger logger)
+    public async Task<string> RunFeeder()
     {
-        if (_famFeederOptions.DbStatusAiCs == null || _famFeederOptions.DbStatusAiCs == "")
+        if (string.IsNullOrEmpty(_famFeederOptions.DbStatusAiCs))
```
[the remark this answers is not in the document]
Comment (TordJoranger) [reply Core/Services/DbStatusFeederService.cs:24]: Agreed, I'll add a comment for now, and its scary to rename fields that are connected to env variables that I dont have knowledge of (Terjes code originally)

[the remark this answers is not in the document]
Comment (TordJoranger) [reply Core/Services/DbStatusFeederService.cs:24]: Actually, its allready a log just under that sort of explains it. so i'll leave it for now

File: Infrastructure/Repositories/DbStatusRepository.cs (hunk 4 above)
Comment (jsoi-eq) [reply Infrastructure/Repositories/DbStatusRepository.cs:47]: Compare mode made it seem more magic, seeing it now it makes more sense, closing it.

Comment (TordJoranger) [reply Infrastructure/Repositories/DbStatusRepository.cs:47]: I think its referring to indexes in the result set. Basically, if index 0 exists, UserName = result at index 0. 
@terjewoldseth  can confirm?

**Review by jsoi-eq — APPROVED** [review]

**Pull request merged**